.circleci/config.yml

```diff
@@ -1,59 +1,74 @@
-version: 2.1
+version: 2.1  # Version of CircleCI config format
 
-orbs:
-  codecov: codecov/codecov@1.1.0
+orbs:  # CircleCI packages ("orbs") we depend on
+  codecov: codecov/codecov@1.1.0  # support for uploading code coverage to codecov
 
 executors:
   my-executor:
     docker:
       - image: humancompatibleai/seals:base
+        # Credentials defined in `docker-hub-creds` context.
+        # We authenticate to avoid Dockerhub per-IP rate limiting of anonymous users.
         auth:
           username: $DOCKERHUB_USERNAME
           password: $DOCKERHUB_PASSWORD
-    working_directory: /benchmark-environments
+    working_directory: /seals
     environment:
       # If you change these, also change ci/code_checks.sh
       SRC_FILES: src/ tests/ setup.py
       TYPECHECK_FILES: src/  # tests often do weird things with types, exclude
       NUM_CPUS: 2  # more CPUs visible but we're throttled to 2, which breaks auto-detect
 
 commands:
+  # Define common function to install dependencies and seals, used in the jobs defined in the next section
   dependencies:
     # You must still manually update the Docker image if any
     # binary (non-Python) dependencies change.
     description: "Check out and update Python dependencies."
     steps:
-      - checkout
+      - checkout  # Check out the code from Git
+
       # Download and cache dependencies
-      # Note you will still need to manually update the Docker image if any
-      # binary (non-Python) dependencies change.
+      # Note the Docker image must still be manually updated if any binary (non-Python) dependencies change.
+
+      # Restore cache if it exists. setup.py defines all the requirements, so we checksum that.
+      # If you need to force an update, then increase the version number `vn-dependencies`.
```

Comment:
Useful to clarify: when would you need to force an update, rather than one happening automatically because of the checksum changing?